packages/blitz-auth/src/client/index.tsx

```diff
@@ -196,25 +197,21 @@ export const useAuthorizeIf = (condition?: boolean, role?: string | Array<string
     throw error
   }
 
-  if (isClient && condition && role && getPublicDataStore().getData().userId && mounted) {
+  if (isClient && condition && roles && getPublicDataStore().getData().userId && mounted) {
     const error = new AuthenticationError()
     error.stack = null!
-    if (!authorizeRole(role, getPublicDataStore().getData().role as string)) {
+    if (!authorizeRole(roles, getPublicDataStore().getData().roles as string[])) {
       throw error
     }
   }
 }
 
-const authorizeRole = (role?: string | Array<string>, currentRole?: string) => {
-  if (role && currentRole) {
-    if (Array.isArray(role)) {
-      if (role.includes(currentRole)) {
-        return true
-      }
+const authorizeRole = (roles?: string | Array<string>, currentRoles?: string[]) => {
```

Comment:
🔨 I have not tested it myself, but the api would need to look like this, to handle a case were a single `currentRole` exists or an array
```ts
const authorizeRole = (role?: string | Array<string>, currentRole?: string | Array<string>) => {
  if (role && currentRole) {
    if (Array.isArray(role) && role.some((r) => currentRole.includes(r))) {
      return true
    } else {
      if (currentRole === role) {
        return true
      }
    }
  }
  return false
}
```